File: src/main/java/com/magina/resource/dao/CloudDiskDAO.java
package com.magina.resource.dao;

import com.magina.resource.domain.CloudDisk;
import com.magina.resource.domain.RequestParam;

import java.util.List;

import com.magina.resource.domain.CloudDiskDTO;

public interface CloudDiskDAO {
    
    List<CloudDisk> listLinks(RequestParam param);
    
    int countLinks(String query);
    
    CloudDisk getLinkById(Long linkId);
    
    void removeLinkById(Long linkId);
    
    void saveLink(CloudDiskDTO cloudDisk);
    
    void updateLink(CloudDiskDTO cloudDisk);
}

File: src/main/kotlin/com/magina/resource/dao/CloudDiskDAO.kt
package com.magina.resource.dao

import com.magina.resource.domain.CloudDisk
import com.magina.resource.domain.RequestParam
import org.apache.ibatis.annotations.Mapper
import org.apache.ibatis.annotations.Param
import com.magina.resource.domain.CloudDiskDTO

@Mapper
interface CloudDiskDAO {
	fun listLinks(param: RequestParam) : List<CloudDisk>
	
	fun countLinks(@Param("query") query: String?) : Int
	
	fun getLinkById(@Param("linkId") linkId: Long) : CloudDisk
	
	fun removeLinkById(@Param("linkId") linkId: Long)
	
	fun saveLink(cloudDisk: CloudDiskDTO)
	
	fun updateLink(cloudDisk: CloudDiskDTO)
}